modules/user/controller.js

```diff
@@ -1,45 +1,84 @@
-    const user = await service.create({ email, password, firstName, lastName })
-    res.send(user)
-}
+  const { email, password, firstName, lastName } = req.body;
+  const user = await service.create({ email, password, firstName, lastName });
+  res.send(user);
+};
 
 export const login = async (req, res) => {
-    const { email, password } = req.body
-    try {
-        const token = await service.login({ email, password })
-        res.send({ token })
-    } catch (error) {
-        res.send(error.message)
-    }
-}
+  const { email, password } = req.body;
+  try {
+    const token = await service.login({ email, password });
+    res.send({ token });
+  } catch (error) {
+    res.send(error.message);
+  }
+};
 
 export const find = async (req, res) => {
-    const users = await User.find()
-    return res.send(users)
-}
+  const users = await User.find();
+  return res.send(users);
+};
 
 export const findById = async (req, res) => {
-    const user = await User.findOne({ _id: req.params.id })
-    // const user = await User.findById(req.id)
-    return res.send(user)
-}
+  const user = await User.findOne({ _id: req.params.id });
+  return res.send(user);
+  // const user = await User.findById(req.id)
+};
 
 export const update = async (req, res) => {
-    res.send(req.body)
-}
+  res.send(req.body);
+};
 
 export const remove = async (req, res) => {
-    // await User.deleteOne({ _id: req.params.id })
-    // res.status(204).send()
+  // await User.deleteOne({ _id: req.params.id })
+  // res.status(204).send()
+  const user = User.findById(req.params.id);
+  if (!user) {
+    throw new Error("user not found");
+  }
+  user.delete();
+  res.status(204).send();
+};
 
-    const user = User.findById(req.params.id)
-    if (!user) {
-        throw new Error('user not found')
-    }
-    user.delete()
-    res.status(204).send()
-}
\ No newline at end of file
+export const addPost = async (req, res) => {
+  const { desc } = req.body;
+  const user = await User.findOne({ _id: req.userId });
+  const post = await service.createPost({ desc, user });
+  res.send(post);
+};
+export const addComment = async (req, res) => {
+  const { desc } = req.body;
+  const user = await User.findOne({ _id: req.userId });
+  const postId = req.params.id;
+  const comment = await service.createComment({ desc, user, postId });
+  res.send(comment);
+};
+export const addPostLike = async (req, res) => {
+  const user = await User.findOne({ _id: req.userId });
+  const refId = req.params.id;
+  const kind = "Post";
+  const like = await service.createLike({ kind, refId, user });
+  await service.addLikePost({ postId: refId });
+  res.send(like);
+};
+export const addCommentLike = async (req, res) => {
+  const user = await User.findOne({ _id: req.userId });
+  const refId = req.params.id;
+  const kind = "Comment";
+  const like = await service.createLike({ kind, refId, user });
+  await service.addLikeComment({ commentId: refId });
+  res.send(like);
```

Comment:
the same applies here

Reply:
pass the user id to the service not the user object, because if we want to call that service => we don't have to query the database, the service should do it for us